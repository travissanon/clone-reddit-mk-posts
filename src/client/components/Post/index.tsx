import React from "react";

// import "./style.scss";

interface IPostProps {
	readonly id: string;
	readonly title: string;
	readonly thumbnail: string;
	readonly author: string;
	readonly favorite: boolean;
	handleFavoriteClick: (postId: string) => void;
}

const Post: React.FC<IPostProps> = ({
	id,
	title,
	thumbnail,
	author,
	favorite,
	handleFavoriteClick,
}) => {
	return (
		<div className="reddit-post" data-id={id}>
			<div className="reddit-post__thumbnail">
				<img src={thumbnail} alt="post thumbnail" />
			</div>
			<div className="reddit-post__stuff">
				<div className="reddit-post__title">
					<h3>{title}</h3>
				</div>
				<div className="reddit-post__author">{author}</div>
			</div>
			<div
				className="reddit-post__favorite"
				onClick={() => handleFavoriteClick(id)}
			>
				<h1>{favorite ? "*" : "x"}</h1>
			</div>
		</div>
	);
};

export default Post;
