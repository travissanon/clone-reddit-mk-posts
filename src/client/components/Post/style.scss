@import url("https://fonts.googleapis.com/css2?family=Open+Sans:wght@400;600&display=swap");

.reddit-post {
	max-width: 720px;
	width: 100%;
	background-color: #f4f4f4;
	border-radius: 8px;
	display: flex;
	flex-direction: row;
	margin: 10px;
	font-family: "Open Sans", sans-serif;

	&__thumbnail {
		margin: 5px;

		img {
			border-radius: 4px;
			height: 80px;
			width: 100px;
		}
	}

	&__title {
		font-size: 16px;
		h3 {
			margin: 0;
			font-weight: 600;
		}
	}

	&__stuff {
		margin: 10px 5px;
		max-width: 500px;
		width: 100%;
	}

	&__author {
		font-size: 14px;
	}
}
