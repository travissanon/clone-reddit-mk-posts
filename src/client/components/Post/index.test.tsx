import * as React from "react";
import * as renderer from "react-test-renderer";

import Post from "./index";

interface IPostProps {
	readonly id: string;
	readonly title: string;
	readonly thumbnail: string;
	readonly author: string;
	readonly favorite: boolean;
	handleFavoriteClick: (postId: string) => void;
}

test("renders content to the page", () => {
	const props: IPostProps = {
		id: "1234",
		title: "This is a post test",
		thumbnail: "",
		author: "tester",
		favorite: true,
		handleFavoriteClick: (postId: string) => {
			null;
		},
	};

	const tree = renderer.create(<Post {...props} />).toJSON();
	expect(tree).toMatchSnapshot();
});
