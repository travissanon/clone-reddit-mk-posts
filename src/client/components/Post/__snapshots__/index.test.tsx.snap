// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content to the page 1`] = `
<div
  className="reddit-post"
  data-id="1234"
>
  <div
    className="reddit-post__thumbnail"
  >
    <img
      alt="post thumbnail"
      src=""
    />
  </div>
  <div
    className="reddit-post__content"
  >
    <div
      className="reddit-post__title"
    >
      <h3>
        This is a post test
      </h3>
    </div>
    <div
      className="reddit-post__author"
    >
      tester
    </div>
  </div>
  <div
    className="reddit-post__favorite"
    onClick={[Function]}
  >
    <img
      alt="favorited post"
      src={Object {}}
    />
  </div>
</div>
`;
